SimG4Core/Application/src/RunManagerMTWorker.cc

```diff
@@ -467,8 +470,9 @@ std::unique_ptr<G4SimEvent> RunManagerMTWorker::produce(const edm::Event& inpevt
   // per-run initialization here by ourselves.
 
   if (!(m_tls && m_tls->threadInitialized)) {
-    edm::LogVerbatim("SimG4CoreApplication") << "RunManagerMTWorker::produce(): stream " << inpevt.streamID()
-                                             << " thread " << getThreadIndex() << " initializing";
+    edm::LogWarning("SimG4CoreApplication")
+        << "RunManagerMTWorker::produce(): stream " << inpevt.streamID() << " thread " << getThreadIndex()
+        << " initializing in the produce(..) method - there is a problem";
```

Comment:
Call to `initializeG4()` in `produce()` should not be an indication of a problem. The framework does not guarantee that all TBB worker threads would call the `beginRun()`. Therefore it is expected that for some threads the `produce()` is the first function in `OscarMTProducer` they call (which is why the code was crafted that way).

Reply:
@makortel , thanks for explanation. In the next PR I will substitute Warning by Verbatim. One of the reason to move initialisation at beginRun(..) is to take out initialisation from the event loop, so this will not affect CPU measurements. For the case, when streams do not coincide with threads we will still have initialisation at the 1st event.